src/nbl/ext/LumaMeter/CLumaMeter.cpp

```diff
@@ -188,12 +187,6 @@ void main()
 	);
 
 	return core::make_smart_refctd_ptr<ICPUSpecializedShader>(
-		core::make_smart_refctd_ptr<ICPUShader>(std::move(shader),ICPUShader::buffer_contains_glsl),
-		ISpecializedShader::SInfo{nullptr, nullptr, "main", asset::ISpecializedShader::ESS_COMPUTE}
-	);
-}
-
-void CLumaMeter::defaultBarrier()
-{
-	COpenGLExtensionHandler::pGlMemoryBarrier(GL_UNIFORM_BARRIER_BIT|GL_SHADER_STORAGE_BARRIER_BIT|GL_BUFFER_UPDATE_BARRIER_BIT);
+		core::make_smart_refctd_ptr<ICPUShader>(std::move(shader), ICPUShader::buffer_contains_glsl, asset::IShader::ESS_COMPUTE, "????"),
+		ISpecializedShader::SInfo{ nullptr, nullptr, "main" });
```

Comment:
use `ostringstream` instead of this ugly sprintf code to create a small compute shader

```c
#define _NBL_GLSL_EXT_LUMA_METER_DISPATCH_SIZE_X_DEFINED_ %d
#define _NBL_GLSL_EXT_LUMA_METER_DISPATCH_SIZE_Y_DEFINED_ %d

#define _NBL_GLSL_EXT_LUMA_METER_BIN_COUNT %d
#define _NBL_GLSL_EXT_LUMA_METER_BIN_GLOBAL_REPLICATION %d
#define _NBL_GLSL_EXT_LUMA_METER_MIN_LUMA_DEFINED_ %d
#define _NBL_GLSL_EXT_LUMA_METER_MAX_LUMA_DEFINED_ %d
#define _NBL_GLSL_EXT_LUMA_METER_MODE_DEFINED_ %d

#define _NBL_GLSL_EXT_LUMA_METER_EOTF_DEFINED_ %s
#define _NBL_GLSL_EXT_LUMA_METER_XYZ_CONVERSION_MATRIX_DEFINED_ %s


#include "nbl/builtin/glsl/ext/LumaMeter/default.comp"
```

Then the only thing you need is just to create an `ICPUShader`, no weird overriden copies, etc.

Reply:
@nipunG314 no more weird shader overrides needed